submissions/antonkottans/friends-app/script.js

```diff
@@ -0,0 +1,595 @@
+    let loadingProgressDots = [];
+
+    const placeLoadingIndicator = () => {
+        const loadingContainer = document.createElement("div");
+        loadingContainer.classList.add("loading-container");
+        const loadingText = document.createElement("span");
+        loadingText.classList.add("loading-text");
+        loadingText.insertAdjacentText("afterbegin", "Loading");
+        loadingProgressDots = [...new Array(pages)].map((_, i) => {
+            const dot = document.createElement("div");
+            dot.classList.add("in-progress", "loading-dot");
+            return dot;
+        });
+        loadingContainer.append(loadingText, ...loadingProgressDots);
+        containerForFriends.appendChild(loadingContainer);
+    };
+
+    placeLoadingIndicator();
+    state.loading = true;
+    friendsStorage.clearFriends();
+
+    Promise.all(
+        [...new Array(pages)].map((_, i) => {
+            return doRecursiveFetch(
+                `https://randomuser.me/api/?inc=${friendFieldsList}&seed=${seed}&page=${
+                    i + 1
+                }&results=${friendsPerPage}`
+            ).then((response) => {
+                loadingProgressDots[i].classList.remove("in-progress");
+                friendsStorage.addFriends(response.results);
+            });
+        })
+    ).then(() => {
+        state.loading = false;
+        updateContentAccordingToActiveFilters();
+    });
+};
+
+let createFriendCard = (friend) => {
+    const createFriendContainer = () => {
+        const friendCard = document.createElement("div");
+        friendCard.classList.add("friend-container", "hidden");
+
+        return friendCard;
+    };
+
+    const addName = (friendCard) => {
+        const friendName = document.createElement("span");
+        friendName.classList.add("name");
+        friendName.textContent = `${friend.name.title}. ${friend.name.first} ${friend.name.last}`;
+        friendCard.appendChild(friendName);
+        return friendCard;
+    };
+
+    const addPortrait = (friendCard) => {
+        const friendPortrait = document.createElement("img");
+        friendPortrait.setAttribute("src", friend.picture.large);
+        friendPortrait.setAttribute("alt", "random friend");
+        friendPortrait.classList.add("portrait");
+        friendCard.appendChild(friendPortrait);
+        return friendCard;
+    };
+
+    const addDOB = (friendCard) => {
+        const friendDOB = document.createElement("span");
+        friendDOB.textContent = `I am ${friend.dob.age} years old.`;
+        friendCard.appendChild(friendDOB);
+        return friendCard;
+    };
+
+    const addEmail = (friendCard) => {
+        const friendEmail = document.createElement("span");
+        friendEmail.textContent = friend.email;
+        friendEmail.classList.add("email");
+        friendCard.appendChild(friendEmail);
+        return friendCard;
+    };
+
+    const addCell = (friendCard) => {
+        const friendCell = document.createElement("span");
```

Comment:
For phone numbers better to use a link with a special value for `href` attribute. [Read more](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#linking_to_telephone_numbers)